docs/specs/code/COIN_PRICE.md

```diff
@@ -0,0 +1,169 @@
+# Abstract
+
+Describe the ability to see and input ADA amount in another currency.
+
+# Proposal
+
+This feature includes two aspects:
+
+## 1. Input
+When the UI contains an ADA amount input, it should let the user input the number in another currency and covert it to ADA amount at the present conversion rate.
+
+## 2. Output
+When the UI shows an ADA amount, additionally it should be able to show the amount in another currency. If the ADA amount is related to an existing transaction, the user will be able to see the price at two different moments:
+-right now
+-at the moment of the transaction.
+
+The following fiat currencies will be supported:
+
+- USD	US dollar	
+- JPY	Japanese yen
+- EUR   Euro
+- CNY	Chinese yuan renminbi
+- KRW	South Korean won
+
+These are the currencies supported by https://exchangeratesapi.io/. See the Architecture section for details.
+
+The following cryptos will be supported:
+- BTC
+- ETH
+
+# Terminology
+
+Frontend: yoroi-frontend.
+
+Backend: yoroi-backend-service, partically the extensions that are required to implement this proposal.
+
+API: third-party services that provide the price data.
+
+Price data: the conversion rates between ADA and all supported currencies.
+
+# Architecture
+
+The frontend refreshes the current ADA price at 15 minute interval by querying the backend.
+
+The backend refreshes the current ADA price at 15 minute interval by querying the API. So the freshness of the price data from the user's perspective is 15 minutes + 15 minutes = 30 minutes. The backend stores the timestamped price data in a historical price database.
+
+At deployment time, a script will be run and it will query the API for historical price data, since the day when ADA began to be traded (Oct 2, 2017), and store the data in the historical price database.
+
+When the frontend shows a transaction it will query the backend for the price at the moment of the transaction. The backend will look into the historical price database and use the price at the closest time point. The frontend will cache the price data in web storage.
+
+To support the price conversions listed above, the backend queries crypto-price API for these pairs:
+
+- ADA-USD
+- BTC-USD
+- ETH-USD
+
+and https://exchangeratesapi.io/ for the exchange rate between USD and every fiat.
+
+# Risk
```

Comment:
Would like to have explicitly the 4 parts (frontend, backend service, data fetcher, api provider) and the risk of each part for this feature &  how we mitigate the risk of hacks in each part